src/eradiate_plugins/bsdfs/rtls.cpp

```diff
@@ -0,0 +1,322 @@
+
+    RTLSBSDF(const Properties &props) : Base(props) {
+        m_f_iso = props.texture<Texture>("f_iso", 0.209741f);
+        m_f_vol = props.texture<Texture>("f_vol", 0.081384f);
+        m_f_geo = props.texture<Texture>("f_geo", 0.004140f);
+
+        /*
+         * Values from: "MODIS BRDF/Albedo Product:
+         * Algorithm Theoretical Basis Document
+         * Version 5.0"
+         */
+        m_h     = props.get<ScalarFloat>("h", 2.0f);
+        m_r     = props.get<ScalarFloat>("r", 1.0f);
+        m_b     = props.get<ScalarFloat>("b", 1.0f);
+        m_flags = BSDFFlags::GlossyReflection | BSDFFlags::FrontSide;
+        dr::set_attr(this, "flags", m_flags);
+        m_components.push_back(m_flags);
+    }
+
+    void traverse(TraversalCallback *callback) override {
+        callback->put_object("f_iso", m_f_iso.get(),
+                             +ParamFlags::Differentiable);
+        callback->put_object("f_vol", m_f_vol.get(),
+                             +ParamFlags::Differentiable);
+        callback->put_object("f_geo", m_f_geo.get(),
+                             +ParamFlags::Differentiable);
+    }
+
+    std::pair<BSDFSample3f, Spectrum> sample(const BSDFContext &ctx,
+                                             const SurfaceInteraction3f &si,
+                                             Float /* position_sample */,
+                                             const Point2f &direction_sample,
+                                             Mask active) const override {
+        MI_MASKED_FUNCTION(ProfilerPhase::BSDFSample, active);
+
+        Float cos_theta_i = Frame3f::cos_theta(si.wi);
+        BSDFSample3f bs   = dr::zeros<BSDFSample3f>();
+
+        active &= cos_theta_i > 0.f;
+        if (unlikely(dr::none_or<false>(active) ||
+                     !ctx.is_enabled(BSDFFlags::GlossyReflection)))
+            return { bs, 0.f };
+
+        bs.wo           = warp::square_to_cosine_hemisphere(direction_sample);
+        bs.pdf          = warp::square_to_cosine_hemisphere_pdf(bs.wo);
+        bs.eta          = 1.f;
+        bs.sampled_type = +BSDFFlags::GlossyReflection;
+        bs.sampled_component = 0;
+
+        UnpolarizedSpectrum value =
+            eval_rtls(si, bs.wo, active) * Frame3f::cos_theta(bs.wo) / bs.pdf;
+
+        return { bs, depolarizer<Spectrum>(value) & (active && bs.pdf > 0.f) };
+    }
+
+    inline const UnpolarizedSpectrum eval_K_iso() const { return 1.0f; }
+
+    inline const UnpolarizedSpectrum eval_K_vol(const Float &cos_theta_i,
+                                                const Float &cos_theta_o,
+                                                const Float &cos_psi,
+                                                const Float &sin_psi,
+                                                const Float &psi) const {
+        return ((dr::Pi<Float> / 2.f - psi) * cos_psi + sin_psi) /
+                   (cos_theta_i + cos_theta_o) -
+               (dr::Pi<Float> / 4.f);
+    }
```

Comment:
Yes, indeed

Reply:
I don't see any of these definitions in drjit. Besides, I a compiler should be able to optimize the division of two compile time constants, shouldn't it?

Reply:
They don't exist in Dr.Jit indeed. I don't know if the compiler can cache these values automatically—I just thought about this because Dr.Jit defines `TwoPi` and `InvPi`.

Reply:
Compared to all the trigo that's going on, it certainly doesn't make much of a difference.